back/app/controllers/web_api/v1/mentions_controller.rb

```diff
@@ -13,10 +13,16 @@ def users
     post = find_post(params[:post_type], params[:post_id])
 
     @users = []
-    @users = MentionService.new.users_from_post(query, post, limit) if post && params[:roles].nil?
+    @users = MentionService.new.users_from_post(query, post, limit, current_user) if post && params[:roles].nil?
 
     nb_missing_users = limit - @users.size
-    @users += find_users_by_query(query, post).limit(nb_missing_users) if nb_missing_users.positive?
+    if nb_missing_users.positive?
+      @users += if current_user
+        find_users_by_query(query, post).where.not(id: current_user.id).limit(nb_missing_users)
+      else
+        find_users_by_query(query, post).limit(nb_missing_users)
+      end
+    end
```

Comment:
Currently, the current user is excluded in both `users_from_post` and here. It can be simplified a bit like that
```ruby
    @users = MentionService.new.users_from_post(query, post, limit) if post && params[:roles].nil?

    nb_missing_users = limit - @users.size
    @users += find_users_by_query(query, post).limit(nb_missing_users) if nb_missing_users.positive?
    @users = @users.excluding(current_user)
```
And then, probably, we need just one test in `back/spec/acceptance/mentions_spec.rb` instead of two.

Reply:
The reason I didn't do it this way, is that this will give `limit - 1` users when the current user is included in `@users`.

The way I do it we will always get `limit n` users _if_ there are enough matches.

Reply:
I'll go with your version, then try to refactor in my 10% time.